swap/tests/alice_and_bob_refund_using_cancel_and_refund_command_timelock_not_expired.rs

```diff
@@ -0,0 +1,108 @@
+pub mod harness;
+
+use harness::alice_run_until::is_xmr_lock_transaction_sent;
+use harness::bob_run_until::is_btc_locked;
+use harness::SlowCancelConfig;
+use swap::protocol::alice::AliceState;
+use swap::protocol::bob::BobState;
+use swap::protocol::{alice, bob};
+
+#[tokio::test]
+async fn given_alice_and_bob_manually_cancel_when_timelock_not_expired_errors() {
+    harness::setup_test(SlowCancelConfig, |mut ctx| async move {
+        let (bob_swap, bob_join_handle) = ctx.bob_swap().await;
+        let swap_id = bob_swap.id;
+        let bob_swap = tokio::spawn(bob::run_until(bob_swap, is_btc_locked));
+
+        let alice_swap = ctx.alice_next_swap().await;
+        let alice_swap = tokio::spawn(alice::run_until(alice_swap, is_xmr_lock_transaction_sent));
+
+        let bob_state = bob_swap.await??;
+        assert!(matches!(bob_state, BobState::BtcLocked { .. }));
+
+        let (bob_swap, bob_join_handle) = ctx
+            .stop_and_resume_bob_from_db(bob_join_handle, swap_id)
+            .await;
+        assert!(matches!(bob_swap.state, BobState::BtcLocked { .. }));
```

Comment:
Are these asserts actually worth anything? Adds a lot of noise to the test for basically no gain (if we are worried that saving something to the DB corrupts the data, we should have more DB tests).

Reply:
See my comment below please - These asserts are not just for `ensuring we saved in the db` but that nothing on the way of stopping/starting the application changes the state!
I don't find them noisy, I actually find them helpful in understanding more complex tests setups as in what to expect from the state at a given point in the test.